### ISSUE TITLE
Tutorial 4: kittes pallet serde import error on both monthly-2021-11-1 and monthly-2021-11-0

### ISSUE THREAD

**Issue by pyramid-songs** [substrate-developer-hub/substrate-docs#576]
_Originally posted by @luatnd in https://github.com/substrate-developer-hub/substrate-docs/issues/516#issuecomment-973129038_

____
For the latest guide, I got this error on the tutorial 4: kittes pallet: if I using `monthly-2021-10` 
```
error[E0463]: can't find crate for `serde`
  --> pallets/kitties/src/lib.rs:42:37
   |
42 |     #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
   |                                        ^^^^^^^^^ can't find crate
   |
   = note: this error originates in the derive macro `Serialize` (in Nightly builds, run with -Z macro-backtrace for more info)
```

My solution is update all dependencies to latest tag following this [commit](https://github.com/sacha-l/substrate-node-template/compare/1655c1241b48667a16a537ebf72c4114ebb104f3...65de1e7fa3682d0ec023a732fa3a4747c8a5013c#)

But... without success :D

____

Also the case for me using the [newly updated](https://github.com/substrate-developer-hub/substrate-docs/issues/556#issuecomment-972922819) `kickstart-substrate` that has `frame-system` `monthly-2021-11-1` '4.0.0-dev'

**Comment by sacha-l** [comment]
I just tested, following these steps:

1.  kickstart https://github.com/sacha-l/kickstart-substrate (and then added the node and template names)
2. cd kitties && cargo build --release
3. Got and error about `sp-io`, which needs to be a `dependency` (not a `dev-dependency`). Changed that in the pallet's Cargo.toml file
4. Pasted in the template code from [this step](https://docs.substrate.io/tutorials/v3/kitties/pt1/#uniqueness-custom-types-and-storage-maps) + added the Currency type for pallet implementation for Runtime in `runtime/src/lib.rs`.
5. Ran cargo build --release
6. No errors

The commit you're referencing isn't the correct one because it's a commit that sets up the submodule node-template that kickstart uses. If you look closely you'll see the `{{node}}` syntax which is what kickstart needs to work as a templating tool. The commits you would have wanted would be from [this PR](https://github.com/substrate-developer-hub/substrate-node-template/pull/259/files).  

One thought is make sure you have `use frame_support::serde::{Deserialize, Serialize};`. Let me know if after reproducing those steps you still get the same errors.

**Comment by pyramid-songs** [comment]
Thanks Sasha-l. I followed your steps with a fresh project & encountered no errors, however the [code step](https://docs.substrate.io/tutorials/v3/kitties/pt1/#uniqueness-custom-types-and-storage-maps) you linked to test doesn't include any instances where serde is included in the code. I should have been clearer in my first post regarding which stage of the tutorial I encountered the error in. It was during the [kittie gender/sex enum declaration](https://docs.substrate.io/tutorials/v3/kitties/pt1/#write-a-custom-type-for-gender) in Part 2:

```
#[derive(Clone, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
  #[scale_info(skip_type_params(T))]
#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
pub enum Gender {
    Male,
    Female,
}
```

I also have the frame_support module loaded:

```
#[cfg(feature = "std")]
use frame_support::serde::{Deserialize, Serialize};
```

Basically, I've finished Part 2 and I'm up to when the tutorial say `Running into difficulties? Check your solution against the completed helper code for this part of the tutorial.`

1 follow-on suggestion:  this would be a great opportunity to have another code 'checkpoint' that exists between 02 and 03; 02 is the 'before part 2' state but 03 is the 'after part 3' state.

**Comment by luatnd** [comment]
You need to add `serde` dependency to your pallet's cargo.toml also.

There's a commit a few days ago switch to use `use frame_support::serde::{Deserialize, Serialize};`
and it remove the `serde` from cargo.toml because it no longer needed.
But I need to add it back to build

**Comment by luatnd** [comment]
FYI, I use the newest template by `kickstart https://github.com/sacha-l/kickstart-substrate`
Also, there're some misc errors I need to fix:
- Above error
- Duplication declaration in contruct_runtime!
   ```
   SubstrateKitties: pallet_kitties,
   Template: pallet_kitties, // I need to comment this out to build
   ```
- Remove test, benchmark, ... in the tutorial source code, because the tutorial also guides me to delete all the files except lib.rs

**Comment by sacha-l** [comment]
@pyramid-songs : Have you tried adding the `serde` dependency back in the `Cargo.toml` file as per @luatnd's recommendation ? I agree, we could have a check point for the code written up until the "Running into difficulties? Check your solution against the completed helper code for this part of the tutorial" line. 

@luatnd : how come you get a duplicate item in `construct_runtime!` ? When you generate the node template with Kickstart,  it should only have `Template: pallet_kitties, `. There's an instruction on renaming the pallet instance. Maybe I'm missing something?

**Comment by pyramid-songs** [comment]
Thanks @sacha-l - adding a `serde` dependency into the pallet Cargo.toml worked and I can compile my code. I was under the impression that frame_support was going to handle the serde functions because of this line:

```
	#[cfg(feature = "std")]
	use frame_support::serde::{Deserialize, Serialize};
```

Definitely worth getting the user to add serde in [step 2 of this section](https://docs.substrate.io/tutorials/v3/kitties/pt1/#write-the-pallet_kitties-scaffold) or editing the default pallet `Cargo.toml` that comes with the kickstart

**Comment by sacha-l** [comment]
👍🏻 Yes - good call.
